engines/dfc_provider/spec/controllers/dfc_provider/api/products_controller_spec.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe DfcProvider::Api::ProductsController, type: :controller do
+  render_views
+
+  let(:user) { create(:user) }
+  let(:enterprise) { create(:distributor_enterprise, owner: user) }
+  let(:product) do
+    create(:simple_product, supplier: enterprise )
+  end
+  let!(:visible_inventory_item) do
+    create(:inventory_item,
+           enterprise: enterprise,
+           variant: product.variants.first,
+           visible: true)
+  end
+
+  describe('.index') do
+    context 'with authorization token' do
+      before do
+        request.env['Authorization'] = 'Bearer 123456.abcdef.123456'
+      end
+
+      context 'with an authenticated user' do
+        before do
+          allow_any_instance_of(DfcProvider::AuthorizationControl)
+            .to receive(:process)
+            .and_return(user)
+        end
+
+        context 'with an enterprise' do
+          before { get :index }
+
+          it 'is successful' do
+            expect(response.status).to eq 200
+          end
+
+          it 'renders the related product' do
+            expect(response.body)
+              .to include("\"DFC:description\":\"#{product.variants.first.name}\"")
+          end
+        end
+
+        context 'without a recorded enterprise' do
+          let(:enterprise) { create(:enterprise) }
```

Comment:
This `#let` call without the bang (eg: `#let!`) will not create or persist this enterprise object until it's directly called, and I'm not sure it's not being called anywhere in this test..?

Reply:
Actually it is called inside the `let!(:visible_inventory_item)` call, so we are good.